config/value.go

```diff
@@ -333,10 +333,41 @@ func derefType(t reflect.Type) reflect.Type {
 	return t
 }
 
+func convertPrimitiveValue(value interface{}, targetType reflect.Type) (interface{}, error) {
+	ret, err := convertVal(value, targetType)
+	if ret == nil {
```

Comment:
These are the three conditions which make err check irrelevant and I can simply jump to 2nd ret != nil check. 
```
// if err is not nil, type is not convertible and we don't need additional type conversion so panic out
if err != nil {
	return ret, err
}
// err is nil, but ret is nil, so we need additional type conversion
if ret != nil {
	return ret, err
}
// just return error
```